packages/imperative/src/cmd/src/response/CommandResponse.ts

```diff
@@ -698,6 +698,32 @@ export class CommandResponse implements ICommandResponseApi {
 
             // Create an instance of the class
             this.mProgressApi = new class {
+                private spinnerIndex = 0;
+                private spinnerInterval: any;
+
+                /**
+                 * Start a spinner
+                 */
+                public startSpinner(pendingText: string): void {
+                    if (this.spinnerInterval == null) {
+                        this.spinnerInterval = setInterval(() => {
+                            process.stdout.write(`\r${pendingText} ${this.mProgressBarSpinnerChars[this.spinnerIndex]}`);
+                            this.spinnerIndex = (this.spinnerIndex + 1) % this.mProgressBarSpinnerChars.length;
+                        }, 100); // eslint-disable-line
+                    }
+                }
+                /**
+                 * Stop a spinner
+                 */
+                public stopSpinner(stopText: string): void {
```

Comment:
And if you don't mind... (picky comment...) 
Could we move these methods (start/stop- spinner) below the endBar method? (and maybe rename `stopSpinner` to `endSpinner`) 😓 

I know, super picky 😅